proto/vendor/README.md

```diff
@@ -1,18 +1,149 @@
+
+**Step 3:**
+
+Whenever you find `import "tendermint/path/to/file.proto"` and the `path/to/file.proto` exists
+in `tendermint-celestia-mods` then add `import "tendermint-celestia-mods/path/to/file.proto"`.
+For example in `abci/types.proto` we added `import "tendermint-celestia-mods/types/types.proto"`
+([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/abci/types.proto#L9)).
+
+**Step 4:**
+
+On any fields that their message type was not added/modified by us, we need to change their type
+to the `tendermint` one. For example in `abci/types.proto` we changed `RequestEcho` types to
+`tendermint.abci.RequestEcho`. It was also needed to add `import "tendermint/abci/types.proto"`
+([link 1](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/abci/types.proto#L21),
+[link 2](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/abci/types.proto#L148)).
+
+If the message type of the field is redefined in `tendermint-celestia-mods` then the one from the
+modifications should be used
+([link](https://github.com/oblique/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/abci/types.proto#L70-L73)).
+Keep in mind that if the type exists in the same file, then full path is not needed.
+
+**Step 5:**
+
+Vendor any message type that contains a modified message type. The best example here is the modified `Data`:
+
+* We need to vendor `tendermint.types.Block` in `tendermint-celestia-mods` because it uses `Data` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/types/block.proto#L9-L14)).
+* Then vendor `tendermint.blockchain.BlockResponse` because it uses `tendermint.types.Block` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/blockchain/types.proto#L9-L11)). Notice that `BlockResponse.block` now has a type of `endermint_celestia_mods.types.Block`.
+* Then vendor `tendermint.blockchain.Message` because it uses `tendermint.blockchain.BlockResponse` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/vendor/tendermint-celestia-mods/blockchain/types.proto#L13-L21)).
+
+**Step 6:**
+
+Modify `build.rs`:
+
+* Add the new `.proto` files in `PROTO_FILES` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/build.rs#L136-L142)).
+* Adjust `CUSTOM_TYPE_ATTRIBUTES` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/build.rs#L58-L62)) and `CUSTOM_FIELD_ATTRIBUTES` ([link](https://github.com/eigerco/lumina/blob/82c51f6ac88fd3662a0f91a0cf19a717986e3470/proto/build.rs#L91-L101)).
```

Comment:
Spitballing: Would it be helpful to be more specific here than just "adjust"? I guess someone reading `build.rs` will figure out that using one of the serde/derive attributes that are defined there may be required, what about having to add a new one?

Reply:
I don't know how to explain "adjust" without creating a whole new chapter. Basically the serialization must match what Celestia is expecting. For example hashes to be serialized as hex string and raw bytes as base64 string.

Reply:
I see, makes sense we cannot explain everything here.